homeassistant/components/statistics/sensor.py

```diff
@@ -598,71 +608,67 @@ def _callable_characteristic_fn(
     # Statistics for numeric sensor
 
     def _stat_average_linear(self) -> StateType:
-        if len(self.states) >= 2:
-            area: float = 0
-            for i in range(1, len(self.states)):
-                area += (
-                    0.5
-                    * (self.states[i] + self.states[i - 1])
-                    * (self.ages[i] - self.ages[i - 1]).total_seconds()
-                )
-            age_range_seconds = (self.ages[-1] - self.ages[0]).total_seconds()
+        if (states := self.states) and len(states) >= 2:
+            ages = self.ages
+            area = 0.0
+            for i in range(1, len(states)):
+                area += 0.5 * (states[i] + states[i - 1]) * (ages[i] - ages[i - 1])
+            age_range_seconds = ages[-1] - ages[0]
             return area / age_range_seconds
         return None
 
     def _stat_average_step(self) -> StateType:
-        if len(self.states) >= 2:
-            area: float = 0
-            for i in range(1, len(self.states)):
-                area += (
-                    self.states[i - 1]
-                    * (self.ages[i] - self.ages[i - 1]).total_seconds()
-                )
-            age_range_seconds = (self.ages[-1] - self.ages[0]).total_seconds()
+        if (states := self.states) and len(states) >= 2:
+            ages = self.ages
+            area = 0.0
+            for i in range(1, len(states)):
+                area += states[i - 1] * (ages[i] - ages[i - 1])
+            age_range_seconds = ages[-1] - ages[0]
             return area / age_range_seconds
         return None
 
     def _stat_average_timeless(self) -> StateType:
         return self._stat_mean()
 
     def _stat_change(self) -> StateType:
-        if len(self.states) > 0:
-            return self.states[-1] - self.states[0]
+        if states := self.states:
+            return states[-1] - states[0]
```

Comment:
Same as above. Here the shady walrus replaces clean readable code without any improvements. Why bother if it does not achieve anything aside confuse the quick reader?

Reply:
https://github.com/home-assistant/core/pull/110438#discussion_r1487894259